### PR TITLE
Improve json filter

### PR DESCRIPTION
**Pull request by Soar360**
I replaced the previous call to IMemberAccessor with GetValueAsync method

### CODE REVIEW

**Comment by sebastienros** [comment]
LGTM. Let me know when you are done. Would have been better with some new unit tests if you actually found an issue, or to show what the new calls to `WriteNullValue` are doing.

**Comment by Soar360** [comment]
I write a new unit test to show why we should call to `WriteNullValue ` .
I'm done,you can merge this pr now, thanks.

File: Fluid.Tests/MiscFiltersTests.cs [hunk 1]
```diff
@@ -630,6 +630,32 @@ public async Task JsonShouldIgnoreStaticMembers()
...
+            var model = new
+            {
+                Id = 1,
+                WithoutIndexable = new DictionaryWithoutIndexableTestObjects(new { }),
+                Bool = true
+            };
+            var options = new TemplateOptions();
+            options.MemberAccessStrategy.Register(model.GetType());
+            var input = FluidValue.Create(model, options);
+
```
Comment (hishamco) [inline Fluid.Tests/MiscFiltersTests.cs:645]: Remove extra line

File: Fluid.Tests/MiscFiltersTests.cs [hunk 2]
```diff
@@ -630,6 +630,32 @@ public async Task JsonShouldIgnoreStaticMembers()
...
+            var result = await MiscFilters.Json(input, new FilterArguments(), new TemplateContext(options));
+            Assert.Equal("{\"Id\":1,\"WithoutIndexable\":null,\"Bool\":true}", result.ToStringValue());
+        }
+
+        private class DictionaryWithoutIndexableTestObjects : ObjectValueBase
```
Comment (hishamco) [inline Fluid.Tests/MiscFiltersTests.cs:650]: Please move this class at the end

File: Fluid/Filters/MiscFilters.cs [hunk 3]
```diff
@@ -673,32 +675,17 @@ private static async ValueTask WriteJson(Utf8JsonWriter writer, FluidValue input
...
-                                value = "circular reference detected.";
-                            }
-
-                            var fluidValue = FluidValue.Create(value, ctx.Options);
-                            if (fluidValue.IsNil())
+                            if(fluidValue is ObjectValue)
                             {
-                                continue;
+                                var value = fluidValue.ToObjectValue();
+                                if(stack.Contains(value)) fluidValue = StringValue.Create("circular reference detected.");
```
Comment (hishamco) [inline Fluid/Filters/MiscFilters.cs:688]:
```csharp
if(stack.Contains(value))
{
    fluidValue = StringValue.Create("Circular reference has been detected.");
}
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Comment by hishamco** [comment]
Looks good, last thing please check why the build fail

**Comment by Soar360** [comment]
> Looks good, last thing please check why the build fail

I have fixed the problem of unit test errors.

**Review by hishamco — APPROVED** [review]
LGTM

**Pull request merged**